manifests/instance.pp

```diff
@@ -141,8 +141,20 @@
     refreshonly => true
   }
 
+  $active_service = $ensure ? {
+    'present' => true,
+    'absent'  => false,
+  }
+
+  $enable_service = $ensure ? {
+    'present' => true,
+    'absent'  => false,
+  }
+
   systemd::unit_file { "github-actions-runner.${instance_name}.service":
     ensure  => $ensure,
+    enable  => $enable_service,
+    active  => $active_service,
```

Comment:
Since version 2.7.0 of the module systemd, this resource can manage the service resource by itself.